lib/cfg.c

```diff
@@ -524,6 +524,27 @@ cfg_dump_processed_config(GString *preprocess_output, gchar *output_filename)
     }
 }
 
+static GString *
+_load_file_into_string(const gchar *fname)
+{
+  FILE  *fd = fopen(fname, "r");
+
+  if (!fd)
+    return NULL;
+
+  gchar buf[1024];
+  gsize res;
+  GString *result = g_string_new("");
+
+  while ((res = fread(buf, 1, sizeof(buf), fd)) > 0)
```

Comment:
Optional:
If the goal is just to read the content, you could use `g_file_get_contents`.
https://developer.gnome.org/glib/stable/glib-File-Utilities.html#g-file-get-contents

Though in the end you need `GString`, though I would still use a library function to read into cstring, then convert to glib, as it would result in cleaner code, and performance is not important here.

Reply:
Thanks, actually I wanted to get such function but my query was not successful, and I did not try that hard.